automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala

```diff
@@ -168,6 +169,56 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
     }
   }
 
+  def stopAndMonitor(googleProject: GoogleProject, clusterName: ClusterName)(implicit webDriver: WebDriver, token: AuthToken): Unit = {
+    Leonardo.cluster.stop(googleProject, clusterName) shouldBe
+      "The request has been accepted for processing, but the processing has not been completed."
+
+    // verify with get()
+    val stoppingCluster = Leonardo.cluster.get(googleProject, clusterName)
+    stoppingCluster.status shouldBe ClusterStatus.Stopping
+
+    // wait until in Stopped state
+    implicit val patienceConfig: PatienceConfig = clusterPatience
+    eventually {
+      val status = Leonardo.cluster.get(googleProject, clusterName).status
+      status shouldBe ClusterStatus.Stopped
+    }
+
+    // Verify notebook error
+    val caught = the [RestException] thrownBy {
+      Leonardo.notebooks.getApi(googleProject, clusterName)
+    }
+    caught.message shouldBe s"""{"statusCode":422,"source":"leonardo","causes":[],"exceptionClass":"org.broadinstitute.dsde.workbench.leonardo.service.ClusterPausedException","stackTrace":[],"message":"Cluster ${googleProject.value}/${clusterName.string} is stopped. Start your cluster before proceeding."}"""
+  }
+
+  def startAndMonitor(googleProject: GoogleProject, clusterName: ClusterName)(implicit webDriver: WebDriver, token: AuthToken): Unit = {
+    Leonardo.cluster.start(googleProject, clusterName) shouldBe
+      "The request has been accepted for processing, but the processing has not been completed."
+
+    // verify with get()
+    val stoppingCluster = Leonardo.cluster.get(googleProject, clusterName)
+    stoppingCluster.status shouldBe ClusterStatus.Starting
+
+    // wait until in Running state
+    implicit val patienceConfig: PatienceConfig = clusterPatience
+    eventually {
+      val status = Leonardo.cluster.get(googleProject, clusterName).status
+      status shouldBe ClusterStatus.Running
+    }
+
+    // TODO cluster is not proxyable yet even after Google says it's ok
```

Comment:
We could. My thinking was to punt this to a separate PR/ticket, since this PR is already too big.

After some digging, I believe the delays is due to the apache proxy in front of Jupyter, not Google. We might also be able to tune this to make it quicker.